source/client-side-encryption/client-side-encryption.rst

```diff
@@ -708,20 +708,44 @@ ClientEncryption
 
       // Creates a new key document and inserts into the key vault collection.
       // Returns the \_id of the created document as a UUID (BSON binary subtype 4).
-      createKey(kmsProvider: String, opts: Optional<DataKeyOpts>): Binary;
+      createKey(kmsProvider: String, opts: Optional<DataKeyOpts>): UUID;
 
       // An alias function equivalent to createKey.
-      createDataKey(kmsProvider: String, opts: Optional<DataKeyOpts>): Binary;
+      createDataKey(kmsProvider: String, opts: Optional<DataKeyOpts>): UUID;
 
       // Decrypts multiple data keys and (re-)encrypts them with a new masterKey, or with their current masterKey if a new one is not given.
       // Returns a RewrapManyDataKeyResult.
       rewrapManyDataKey(filter: Document, opts: Optional<RewrapManyDataKeyOpts>): RewrapManyDataKeyResult;
 
+      // Removes the key document with the given \_id from the key vault collection.
+      // Returns the result of the internal deleteOne() operation on the key vault collection.
+      deleteKey(id: UUID): DeleteResult;
```

Comment:
In this case, are drivers permitted to be flexible in the argument type? If they support a native UUID type, can they accept that and internally prepare it as a Binary with subtype `0x04`? If not, I think this should also require a Binary argument, with a comment requiring the implementation to validate it as a subtype `0x04`.

Reply:
If you want to allow flexibility (no objections there), I think that should be communicated in the comments. And if this is a general point, perhaps you can add a heading higher up explaining that language UUID types and Binary subtype `0x04` may be used interchangeably in APIs.

Reply:
I do not know if that is within the scope of this PR. That appears to be more a property of driver implementation of UUID per current UUID spec rather than something specific to the Key Management API. I will defer to the existing interface (`Binary` return value from `createDataKey`) for now, but such flexibility is _already_ permitted by the spec, then there should be no changes necessary here. If it is _not_ already permitted by the spec, I do not know if it is desirable to begin with. @kevinAlbs Thoughts?

Reply:
This comment thread is marked as "Outdated", but it looks like @kevinAlbs' previous comment warrants some changes to permit drivers to use a native UUID type and Binary with subtype `0x04` interchangeably.